libcxx/src/atomic.cpp

```diff
@@ -178,6 +178,7 @@ _LIBCPP_EXPORTED_FROM_ABI void __cxx_atomic_notify_one(__cxx_atomic_contention_t
 _LIBCPP_EXPORTED_FROM_ABI void __cxx_atomic_notify_all(__cxx_atomic_contention_t const volatile* __location) {
   __libcpp_contention_notify(&__libcpp_contention_state(__location)->__contention_state, __location, false);
 }
+// This function is never used, but still exported for ABI compatibility.
```

Comment:
Is this because this is handled by `__poll_or_get_monitor` now?

Reply:
Yes, exactly. This overload of `__libcpp_atomic_monitor` was used to get the "monitor" value in case the `ptrdiff_t` of the semaphore could be waited on directly. And this was the reason for the original ABA bug, where this could happen:

1. `__libcpp_atomic_monitor` returns "1" (i.e. the semaphore has the value 1)
2. another thread swoops in, takes the "1" down to "0"
3. "our" poll function returns "false", because there is nothing left to `try_acquire`
4. another thread releases the semaphore (increases the value from "0" to "1")
5. `atomic_wait` now waits on the "1", resulting in a deadlock.

Reply:
And now, `__poll_or_get_monitor` makes sure that in this case, the "monitor" value is always "0" (because `return __poll_(__monitor);` sets `__monitor` to "0" if it returns false).